### PR TITLE
Continuous Integration Fixes and Updates

### PR DESCRIPTION
**Pull request by jennmald**
The GitHub workflows is now updated to include Podman integration and Python version 3.9. There are also updates to the vagrant file to install SRW, Shadow, and PyTorch packages. Next steps will be to update documentation to remove the PyTorch installation step.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -32,46 +31,42 @@ Vagrant.configure("2") do |config|
...
     wget -P /tmp https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
     bash /tmp/Miniconda3-latest-Linux-x86_64.sh -b -p /home/vagrant/miniconda3
     rm /tmp/Miniconda3-latest-Linux-x86_64.sh
     /home/vagrant/miniconda3/bin/conda init --system
     /home/vagrant/miniconda3/bin/conda update conda -y
-
     # create a conda environment for development
     /home/vagrant/miniconda3/bin/conda create -y -n dbs python=3.8
+    # install pytorch and dependencies
+    /home/vagrant/miniconda3/bin/conda install -y pytorch torchvision torchaudio cpuonly -c pytorch
```
Comment (jklynch) [inline Vagrantfile:48]: I think the pytorch, srw, and shadow install steps need `-n dbs` or they will install into the base environment.

File: Vagrantfile [hunk 2]
```diff
@@ -22,7 +22,6 @@ Vagrant.configure("2") do |config|
     apt install -y python3-pip
     # install X11 for matplotlib
     apt install -y xserver-xorg-core x11-utils x11-apps
-
```
Comment (mrakitin) [inline Vagrantfile:25]: Why are these removed? It was easier to read. Are the blank spaces causing any issues with running Vagrant?

File: Vagrantfile (hunk 1 above)
Comment (mrakitin) [reply Vagrantfile:48]: I agree with that. Alternatively, the conda env can be activated first, then you will not have to specify the env name.

Also, you may want to source `conda.sh` so that the conda-related env variables are set, and you could use just `conda` command without the need to specify the full path.

```bash
. /home/vagrant/miniconda3/etc/profile.d/conda.sh
```

**Review by mrakitin — COMMENTED** [review]
Looks good, have a couple of suggestions.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -1,58 +1,88 @@
...
-  DBS_tests:
-    name: Test DBS with Python ${{ matrix.python-version }}
+  build:
+    name: Test deep-beamline-sim with ${{ matrix.docker-binary }} and Python ${{ matrix.python-version }}
     runs-on: ubuntu-latest
     strategy:
-      matrix: 
-        python-version: [3.7,3.8]
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
```
[the remark this answers is not in the document]
Comment (jennmald) [reply .github/workflows/tests.yml]: good to know! Just fixed this

File: Vagrantfile (hunk 1 above)
Comment (jennmald) [reply Vagrantfile]: fixed, thank you!

File: Vagrantfile (hunk 2 above)
Comment (jennmald) [reply Vagrantfile]: I'm not sure what happened, I committed them back in.

Comment (mrakitin) [reply Vagrantfile]: Thanks!

**Review by mrakitin — APPROVED** [review]

File: Vagrantfile (hunk 1 above)
Comment (jklynch) [reply Vagrantfile]: my preference in a Vagrantfile is `-n <conda env name>` since activating the conda environment could affect later steps, and I don't have to remember to deactivate it

**Review by jklynch — APPROVED** [review]
